### PR TITLE
feat: customize PR sizing

### PR DESCRIPTION
**Pull request by waltergalvao**
## Release Notes

- **New Features**
	- Introduced enhanced pull request tracking with new metrics: `changedFilesCount`, `linesAddedCount`, and `linesDeletedCount`.
	- Added functionality for managing workspace settings, including a new mutation for updating settings.
	- New components for pull request size settings and a dedicated settings page for pull requests.
	- Enhanced user interface for pull request size customization with additional navigation options.
  
- **Bug Fixes**
	- Improved error handling and data retrieval in pull request size labeling automation.

- **Documentation**
	- Updated GraphQL schema to include new types and mutations related to workspace settings and pull request tracking.
	- Revised documentation to reflect the availability of pull request size customization features.

- **Chores**
	- Cleaned up routing to integrate new settings pages and components effectively.

### CODE REVIEW

File: apps/api/src/app/github/services/github-pull-request.service.ts [hunk 1]
```diff
@@ -204,16 +223,75 @@ const fetchPullRequest = async (
...
+      {
+        nodeId: pullRequestId,
+        cursor: fileQuery.pageInfo.endCursor,
+      }
+    );
+
+    files.push(...response.node.files.nodes);
+    fileQuery.pageInfo = response.node.files.pageInfo;
+  }
+
```
[the remark this answers is not in the document]
Comment (waltergalvao) [reply apps/api/src/app/github/services/github-pull-request.service.ts:283]: Good catch

File: apps/web/src/app/automations/settings/pr-size-labeler/components/form-pr-size-labeler-settings/form-pr-size-labeler-settings.tsx [hunk 2]
```diff
@@ -13,49 +25,68 @@ export const FormPrSizeLabelerSettings = ({
...
+        <Button
+          variant="default"
+          component={Link}
+          to="/settings/pull-request"
+          target="_blank"
+          w="fit-content"
+          rightSection={<IconExternalLink stroke={1.5} size={14} />}
+        >
+          Settings
+        </Button>
```
[the remark this answers is not in the document]
Comment (waltergalvao) [reply apps/web/src/app/automations/settings/pr-size-labeler/components/form-pr-size-labeler-settings/form-pr-size-labeler-settings.tsx:88]: Isn't that the case only for external links?

File: apps/web/src/app/automations/settings/pr-size-labeler/page.tsx [hunk 3]
```diff
@@ -54,6 +54,7 @@ export const AutomationPrSizeLabelerPage = () => {
             color: settings?.labels?.huge?.color || "#ff8787",
           },
         },
+        repositories: [],
```
[the remark this answers is not in the document]
Comment (waltergalvao) [reply apps/web/src/app/automations/settings/pr-size-labeler/page.tsx:57]: That's actually on purpose. We should reset repositories when they save new label settings.

File: apps/web/src/app/settings/pull-request-settings/size/components/form-pull-request-size-settings.tsx [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+      </Stack>
+      <Divider my="sm" />
+      <Stack p="md">
+        <Title order={5}>Ignore Files</Title>
+        <TagsInput
+          label="Glob Patterns"
+          description="Files matching these patterns will be ignored when calculating total changes."
+          value={["package-lock.json", "yarn.lock"]}
+          {...form.getInputProps(`settings.pullRequest.size.ignorePatterns`)}
+        />
```
[the remark this answers is not in the document]
Comment (waltergalvao) [reply apps/web/src/app/settings/pull-request-settings/size/components/form-pull-request-size-settings.tsx:66]: Oops, this was a test leftover, good catch.

**Review by sweetrdev — APPROVED** [review]

**Pull request merged**